### PR TITLE
Test/find on shelf

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-247

#### Description
This PR introduces a new story - periodical - for the find on shelf component, a cypress test for all the component stories (default + periodical), alongside a new fixture. 

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions
n/a

### CODE REVIEW

File: src/components/find-on-shelf/find-on-shelf.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              reservable: true,
+              available: true,
+              reservations: 48
+            }
+          ]
+        });
+      })
+      .as("Find on shelf availability");
+    cy.visit("/iframe.html?id=components-find-on-shelf-modal--periodical");
+    cy.contains("button:visible", "FIND PÅ HYLDEN").click();
```
Comment (kasperg) [inline src/components/find-on-shelf/find-on-shelf.test.ts]: Consider whether the text "FIND PÅ HYLDEN" needs to be in all caps. In general I would expect such stylistic casing to be controlled with CSS.

Consider whether this text "FIND PÅ HYLDEN" should be in English as it has been decided that this is the default language - also for controls in Storybook.

File: src/components/find-on-shelf/find-on-shelf.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  it("Contains two dropdowns when it's a periodical", () => {
+    cy.get("select").contains("2022");
+    cy.get("select").contains("29");
+  });
+
+  it("Doesn't contain two dropdowns when it's not a periodical", () => {
+    cy.visit("/iframe.html?id=components-find-on-shelf-modal--default");
+    cy.contains("button:visible", "FIND PÅ HYLDEN").click();
+    cy.get("select").should("not.exist");
+  });
```
Comment (kasperg) [inline src/components/find-on-shelf/find-on-shelf.test.ts]: Please create different tests for different cases: 1) Find on self default 2) Find on shelf for periodicals

File: src/components/find-on-shelf/find-on-shelf.test.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  it("Contains text with the sum of all available branches", () => {
+    cy.contains("biblioteker har materialet");
+  });
```
Comment (kasperg) [inline src/components/find-on-shelf/find-on-shelf.test.ts]: Please include the actual number in the test.

This is the closest we can get to testing the counting logic.

Also note the language.

File: src/components/find-on-shelf/find-on-shelf.test.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  it("Shows the main branch first", () => {
+    cy.get("summary").eq(0).contains("Hovedbiblioteket");
+  });
+
+  it("Contains a summary for the main branch with no available speciments", () => {
+    cy.get("summary").contains("Hovedbiblioteket");
+    cy.get("summary").contains("unavailable");
```
Comment (kasperg) [inline src/components/find-on-shelf/find-on-shelf.test.ts]:
```suggestion
    cy.get("summary")
      .contains("Hovedbiblioteket")
      .parent()
      .contains("unavailable");
```

Consider whether you need to chain the `contains()` assertions to check whether the item is actually unavailable at the main library branch or just unavailable at some library branch.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: Good stuff here. I have a few things for you to look into.

In general the whole `selectedPeriodical` arg with control type null etc. seems a bit odd. Perhaps it is just me who has not seen this pattern before.

File: src/components/find-on-shelf/find-on-shelf.test.ts (hunk 1 above)
Comment (Adamik10) [reply src/components/find-on-shelf/find-on-shelf.test.ts]: Great points! Fixed both ;)

File: src/components/find-on-shelf/find-on-shelf.test.ts (hunk 2 above)
Comment (Adamik10) [reply src/components/find-on-shelf/find-on-shelf.test.ts]: Very good thinking!

File: src/components/find-on-shelf/find-on-shelf.test.ts (hunk 4 above)
Comment (Adamik10) [reply src/components/find-on-shelf/find-on-shelf.test.ts]: That is very correct. Mine was just checking any for unavailable. 
🥇 🏆

**Comment by Adamik10** [comment]
To follow this PR, please go to PR #187 in the upstream fork

**Pull request closed without merging**